web/src/main/java/com/graphhopper/http/GHBaseServlet.java

```diff
@@ -65,33 +71,33 @@ protected void writeJson(HttpServletRequest req, HttpServletResponse res, JSONOb
             }
 
             if (debug)
-                writeResponse(res, callbackName + "(" + json.toString(2) + ")");
+                writeResponse(res, callbackName + "(" + objectMapper.writeValueAsString(json) /*.toString(2)*/ + ")");
             else
-                writeResponse(res, callbackName + "(" + json.toString(0) + ")");
+                writeResponse(res, callbackName + "(" + objectMapper.writeValueAsString(json) /*.toString(0)*/ + ")");
 
         } else {
             res.setContentType("application/json");
             if (debug)
-                writeResponse(res, json.toString(2));
+                writeResponse(res, objectMapper.writeValueAsString(json) /*.toString(2)*/);
```

Comment:
Here we would need an additional "pretty" objectMapper: 

ObjectMapper prettyOM = new ObjectMapper();
prettyOM.enable(SerializationFeature.INDENT_OUTPUT);